contracts/DisputeManager.sol

```diff
@@ -0,0 +1,129 @@
+        Accepted,
+        Rejected,
+        Cancelled
+    }
+
+    struct Dispute {
+        uint256 disputeId;
+        address indexer;                  // indexer address
+        address fisherman;                // fisherman address
+        uint256 stakingAmount;            // indexer staking amount
+        uint256 depositAmount;            // fisherman deposit amount
+        bytes32 metadata;                 // IPFS address of metadata payload	
+        bytes32 deploymentId;             // project deployment id
+        DisputeType dtype;                // dispute type (POI or Query)
+        uint256 expireEra;                // dispute expired era number
+        DisputeState state;               // dispute state, defult ongoing (ongoing, accept, reject, cancelled)
+    }
+
+    ISettings public settings;
+    uint256 public nextDisputeId;
+    mapping(uint256 => Dispute) public disputes;
+    mapping(address => uint256) public disputeIdByIndexer;
+
+    event DisputeOpen(uint256 indexed disputeId, address fisherman, address indexer, bytes32 metadata, DisputeType _type);
+
+    function initialize(ISettings _settings) external initializer {
+        __Ownable_init();
+
+        settings = _settings;
+        nextDisputeId = 1;
+    }
+
+    function createDispute(
+        address _indexer, 
+        bytes32 _deploymentId; 
+        uint256 _deposit, 
+        bytes32 _metadata, 
+        DisputeType _type
+    ) external {
+        require(disputeIdByIndexer[_indexer] == 0, 'indexer already on a dispute')
+        require(_deposit >= minimumDeposit, 'Not meet the minimum deposit');
+        IERC20(settings.getSQToken()).safeTransferFrom(msg.sender, address(this), _deposit);
+
+        // initial the channel
+        Dispute storage dispute = disputes[nextDisputeId];
+        dispute.disputeId = nextDisputeId;
+        dispute.indexer = _indexer;
+        dispute.fisherman = msg.sender;
+        dispute.stakingAmount = IStaking(settings.getStaking()).getAfterDelegationAmount(_indexer, _indexer);
+        dispute.depositAmount = _deposit;
+        dispute.metadata = _metadata;
+        dispute.deploymentId = _deploymentId;
+        dispute.dtype = _type;
+        dispute.expireEra = IEraManager(settings.getEraManager()).eraNumber()+2;
+        dispute.state = DisputeState.Ongoing;
+
+        disputeIdByIndexer[_indexer] = nextDisputeId;
+
+        emit DisputeOpen(nextDisputeId, msg.sender, _indexer, _metadata, _type);
+        nextDisputeId++;
+    }
+
+    function extendDispute(uint256 disputeId, uint256 newExpireEra) external onlyOwner {
+        Dispute storage dispute = disputes[disputeId];
+        require(newExpireEra > dispute.expireEra, 'invalid newExpireEra');
+        require(dispute.state == DisputeState.Ongoing, 'dispute already finalized');
+
+        dispute.expireEra = newExpireEra;
+    }
+
+    function finalizeDispute(uint256 disputeId, DisputeState state, uint256 newStake, uint256 newDeposit) external onlyOwner {
```

Comment:
The caller should be multi-sign wallet, in that case may need to config in the contract